x-pack/test/api_integration/apis/uptime/graphql/monitor_states.ts

```diff
@@ -177,5 +178,58 @@ export default function({ getService }: FtrProviderContext) {
         });
       });
     });
+
+    describe(' test status filter', async () => {
+      const upMonitorId = 'up-test-id';
+      const downMonitorId = 'down-test-id';
+      const mixMonitorId = 'mix-test-id';
+      before('generate three monitors with up, down, mix state', async () => {
+        await getService('esArchiver').load('uptime/blank');
+
+        const es = getService('legacyEs');
+
+        const observer = {
+          geo: {
+            name: 'US-East',
+            location: '40.7128, -74.0060',
+          },
+        };
+
+        // Generating three monitors each with two geo locations,
+        // One in a down state ,
+        // One in an up state,
+        // One in a mix state
+
+        dateRangeStart = new Date().toISOString();
+
+        await makeChecksWithStatus(es, upMonitorId, 1, 4, 1, {}, 'up');
+        await makeChecksWithStatus(es, upMonitorId, 1, 4, 1, { observer }, 'up');
+
+        await makeChecksWithStatus(es, downMonitorId, 1, 4, 1, {}, 'down');
+        await makeChecksWithStatus(es, downMonitorId, 1, 4, 1, { observer }, 'down');
+
+        await makeChecksWithStatus(es, mixMonitorId, 1, 4, 1, {}, 'up');
+        await makeChecksWithStatus(es, mixMonitorId, 1, 4, 1, { observer }, 'down');
+
+        dateRangeEnd = new Date().toISOString();
+      });
+
+      after('unload heartbeat index', () => getService('esArchiver').unload('uptime/blank'));
+
+      it('should return all monitor when no status filter', async () => {
+        const nonSummaryRes = await getMonitorStates({});
+        expect(nonSummaryRes.monitorStates.summaries.length).to.eql(3);
+      });
+
+      it('should return a monitor with mix state if check status filter is down', async () => {
+        const nonSummaryRes = await getMonitorStates({ statusFilter: 'down' });
+        expect(nonSummaryRes.monitorStates.summaries.length).to.eql(2);
+      });
+
+      it('should not return a monitor with mix state if check status filter is up', async () => {
+        const nonSummaryRes = await getMonitorStates({ statusFilter: 'up' });
+        expect(nonSummaryRes.monitorStates.summaries.length).to.eql(1);
+      });
+    });
```

Comment:
@andrewvc added test as you described :)

Reply:
We need one more tweak here, instead of checking the counts lets check that the monitor IDs we expect are coming back. We'd hate to find out we're returning 1 of the wrong monitor!

Reply:
done.